pipeline/matching/rankers/quantity_test.py

```diff
@@ -27,9 +27,58 @@ def test_example():
         Match(users={"A", "B"}),
         Match(users={"A", "C"}),
     ]
-    # TODO: Uncomment this assertion and delete the proposed assert
-    # assert actual == expected
-    assert actual == proposed
+    assert actual == expected
 
 
-# TODO: Add more test cases for your logic
+def test_example2():
+    inp = MatchingInput(
+        community="test",
+        release="2022-04-01",
+        users=[],
+        recent_matches=[],
+    )
+    proposed = [
+        Match(users={"A", "B", "C"}),  # total num of matches is 5
+        Match(users={"A", "C", "E"}),  # total is 6
+        Match(users={"E", "D", "F"}),  # total is 4
+        Match(users={"X", "Y", "Z"}),  # total is 3
+    ]
+    # A:2 C:2 E:2 BDFXYZ:1
+    ranker = QuantityRanker()
+    actual = list(ranker.rank(inp, proposed))
```

Comment:
Per the previous comment about generators having values taken out, you can simulate that in your unit test by casting the list of proposed matches to `iter()`. After that, this test should fail and you can try to fix the issue.

```python
actual = list(ranker.rank(inp, iter(proposed)))
```

Reply:
Typecasted proposed to an iterable.